File: app/src/main/java/com/lostincontext/mainscreen/MainScreenComponent.java
package com.lostincontext.mainscreen;

import com.lostincontext.application.ApplicationComponent;
import com.lostincontext.commons.FragmentScope;

import dagger.Component;

@FragmentScope
@Component(modules = MainScreenPresenterModule.class, dependencies = ApplicationComponent.class)
public interface MainScreenComponent {

    void inject(MainActivity activity);
}

File: app/src/main/java/com/lostincontext/mainscreen/MainScreenComponent.kt
package com.lostincontext.mainscreen

import com.lostincontext.application.ApplicationComponent
import com.lostincontext.commons.FragmentScope

import dagger.Component

@FragmentScope
@Component(modules = arrayOf(MainScreenPresenterModule::class),
           dependencies = arrayOf(ApplicationComponent::class))
interface MainScreenComponent {

    fun inject(fragment : MainScreenFragment)
}